README.md

```diff
@@ -142,11 +142,26 @@ More detailed documentation is available here: https://oneapi-src.github.io/unif
 #### OS memory provider
 
 A memory provider that provides memory from an operating system.
-It supports two types of memory mappings
+
+OS memory provider supports two types of memory mappings (set by the `visibility` parameter):
 1) private memory mapping (`UMF_MEM_MAP_PRIVATE`)
 2) shared memory mapping (`UMF_MEM_MAP_SHARED` - supported on Linux only yet)
 
-If the shared memory mapping is used then an anonymous file descriptor for memory mapping is created using:
+There are available two mechanisms for the shared memory mapping:
+1) a named POSIX shared memory object (used if the `shm_name` parameter is not NULL) or
+2) an anonymous file descriptor (used if the `shm_name` parameter is NULL)
+
+A named POSIX shared memory object is used only if the `shm_name` parameter is not NULL, otherwise an anonymous file descriptor is used.
+
+A name of a POSIX shared memory object should be of the form `/somename`; that is, a null-terminated string of up to NAME_MAX (i.e., 255) characters consisting of an initial slash, followed by one or more characters, none of which are slashes (see the shm_open(3) manual). The POSIX shared memory object should be created by a user as a file `/dev/shm/somename` with the appropriate permissions and size in the following way:
```

Comment:
> The POSIX shared memory object should be created by a user as a file `/dev/shm/somename` with the appropriate permissions and size

Why does the user need to create a file?

Reply:
I added an option to create an SHM file if open() fails.
Done.